### PR TITLE
Template Data Download

### PR DESCRIPTION
**Pull request by nspool**
Adds a `Template Data` menu item. On this screen, select the Area Type to display a list of areas of that type. Select a type to add it to the filter. Click the minus sign next to a filter to remove. Download Sheets button will download a file named `template.zip` containing necessary files.

Stack is available here: http://nrmn-template-data-download.dev.aodn.org.au/data/extract

### CODE REVIEW

File: api/src/main/java/au/org/aodn/nrmn/restapi/controller/SiteController.java [hunk 1]
```diff
@@ -51,6 +52,22 @@
         return siteRepository.findAll();
     }
 
+    @GetMapping(path = "/siteCodes")
+    public ResponseEntity<List<String>> getAllSiteAreas() {
+        return ResponseEntity.ok(siteRepository.findAllSiteCodes().stream().collect(Collectors.toList()));
```
Comment (jonescc) [inline api/src/main/java/au/org/aodn/nrmn/restapi/controller/SiteController.java:57]: Why not just return a list in the repository methods and not bother converting to one here.

**Review by jonescc — COMMENTED** [review]
Should the drop downs be sorted?

**Comment by nspool** [comment]
@jonescc implemented feedback and redeployed test stack. Ready for review again.

**Review by jonescc — APPROVED** [review]
Looks pretty good to me.

**Pull request merged**